### PR TITLE
feat(logrus): add support for WithContext method

### PR DESCRIPTION
**Pull request by damianopetrungaro**
Add support for `WithContext` method from logrus package

### CODE REVIEW

**Comment by damianopetrungaro** [comment]
@hexdigest any update on this?

File: generator/types.go [hunk 1]
```diff
@@ -312,6 +324,26 @@ func (m Method) HasParams() bool {
...
+// GetContext returns the variable name of the context param
+func (m Method) GetContext() string {
+	for _, p := range m.Params {
+		if p.Type == "context.Context" {
+			return p.Name
+		}
+	}
+	return ""
+}
+
```
Comment (hexdigest) [inline generator/types.go:346]: Gowrap already has this:
* PTAL at the AcceptsContext method helper
* If method accepts context the context argument is always called "ctx"

File: templates_tests/interface_with_logrus_test.go [hunk 2]
```diff
@@ -104,3 +114,25 @@ func TestTestInterfaceWithLogrus_F(t *testing.T) {
...
+	return []logrus.Level{
+		logrus.TraceLevel,
+		logrus.DebugLevel,
+		logrus.InfoLevel,
+		logrus.WarnLevel,
+		logrus.ErrorLevel,
+		logrus.FatalLevel,
+		logrus.PanicLevel,
+	}
+}
```
Comment (hexdigest) [inline templates_tests/interface_with_logrus_test.go:138]: Based on the fact the context.Context var is always named "ctx" can we just get rid of this and check that there's no "ctx" var in the callingRecord above?

**Review by hexdigest — CHANGES_REQUESTED** [review]

**Comment by hexdigest** [comment]
@damianopetrungaro first of all thank you for your contribution. Please take a look at my review.

**Pull request closed without merging**